web/packages/core/src/ruffle-player.ts

```diff
@@ -677,6 +685,26 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * Called when entering / leaving fullscreen on non-Safari
+     */
+    private fullScreenChange(): void {
+        // check if we have just stopped being full screen
+        if (document.fullscreenElement === null) {
+            this.instance?.set_fullscreen(false);
+        }
+    }
+
+    /**
+     * Called when entering / leaving fullscreen on Safari
+     */
+    private webKitFullScreenChange(): void {
```

Comment:
This is a bit of code duplication.
It could be done using a single event listener that handles both `document.fullscreenElement` and `document.webkitFullscreenElement`.